packages/@o3r/configuration/schematics/add-config/index.ts

```diff
@@ -0,0 +1,327 @@
+import {
+  apply,
+  chain,
+  MergeStrategy,
+  mergeWith,
+  move,
+  noop,
+  renameTemplateFiles,
+  Rule,
+  SchematicContext,
+  strings,
+  template,
+  Tree,
+  url
+} from '@angular-devkit/schematics';
+import {
+  applyEsLintFix,
+  getBlockStatementsFromString,
+  getClassElementsFromString,
+  getImplementsExpressionWithTypeArguments,
+  getO3rComponentInfo,
+  getParametersDeclarationFromString,
+  isO3rClassComponent,
+  isO3rClassDecorator,
+  sortClassElement
+} from '@o3r/schematics';
+import { getLibraryNameFromPath } from '@o3r/schematics';
+import { insertImport } from '@schematics/angular/utility/ast-utils';
+import { applyToUpdateRecorder, Change } from '@schematics/angular/utility/change';
+import { basename, dirname } from 'node:path';
+import * as ts from 'typescript';
+import type { NgAddConfigSchematicsSchema } from './schema';
+
+/**
+ * Add configuration to an existing component
+ *
+ * @param options
+ */
+export function ngAddConfig(options: NgAddConfigSchematicsSchema): Rule {
+  return (tree: Tree, context: SchematicContext) => {
+    const { name, selector } = getO3rComponentInfo(tree, options.path) || {};
+
+    if (!name || !selector) {
+      throw new Error(`Unable to retrieve component information at this path ${options.path}`);
```

Comment:
component information => component name or selector (I think the error should be the most accurate possible)